lighthouse-core/gather/gatherers/seo/font-size.js

```diff
@@ -109,28 +183,43 @@ function getNodeTextLength(node) {
 /**
  * @param {Driver} driver
  * @param {LH.Crdp.DOM.Node} node text node
- * @returns {WebInspector.CSSStyleDeclaration}
+ * @returns {Promise<FailingNodeData['cssRule']|undefined>}
  */
-function getFontSizeSourceRule(driver, node) {
-  return driver.sendCommand('CSS.getMatchedStylesForNode', {nodeId: node.nodeId})
-    .then(matchedRules => getEffectiveRule(FONT_SIZE_PROPERTY_NAME, node, matchedRules));
+async function fetchSourceRule(driver, node) {
+  const matchedRules = await driver.sendCommand('CSS.getMatchedStylesForNode', {
+    nodeId: node.nodeId,
+  });
+  const sourceRule = getEffectiveFontRule(matchedRules);
+  if (!sourceRule) return undefined;
+
+  return {
+    type: sourceRule.type,
+    range: sourceRule.range,
+    styleSheetId: sourceRule.styleSheetId,
+    parentRule: sourceRule.parentRule && {
+      origin: sourceRule.parentRule.origin,
+      selectors: sourceRule.parentRule.selectors,
+    },
+  };
 }
 
 /**
  * @param {Driver} driver
  * @param {LH.Artifacts.FontSize.DomNodeWithParent} node text node
- * @returns {Promise<?{fontSize: number, textLength: number, node: LH.Artifacts.FontSize.DomNodeWithParent}>}
+ * @returns {Promise<?FailingNodeData>}
  */
-function getFontSizeInformation(driver, node) {
-  return driver.sendCommand('CSS.getComputedStyleForNode', {nodeId: node.parentId})
+function fetchComputedFontSize(driver, node) {
```

Comment:
any reason why we are not asyncing this function?

Reply:
oh no real reason other than I didn't touch any of the logic, small diff whens don't mean much in this PR though 😆